_source/_posts/2022-02-21-build-deploy-node-app-heroku.md

```diff
@@ -0,0 +1,335 @@
+- "Learn how to build and deploy a @nodejs app to @heroku 🔥"
+image: blog/build-deploy-node-app-heroku/cover-image.jpg
+type: conversion
+github: https://github.com/oktadev/okta-node-heroku-app-example
+---
+
+Heroku is a platform as a service that supports many languages.  Initially, it supported only Ruby sites but now supports various languages, including JavaScript with Node.js.  Heroku also has docker support so that you can deploy just about anything to it.  
+
+This tutorial will teach you how to build a small application using the Express framework for Node.js.  You will then secure that application using Okta by creating an application in Okta for web applications and integrating the Okta OIDC middleware with your application.  Finally, you will learn how to deploy the application to Heroku, update your Okta application for your Heroku site, and set the environment variables for your app in Heroku.
+
+This tutorial uses the following technologies but doesn't require any prior experience:
+
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [Node.js](https://nodejs.org/en/)
+- An [Okta Developer Account](https://developer.okta.com/) (free forever, to handle your OAuth needs)
+- [Heroku Account](https://signup.heroku.com/)
+- A [repository on GitHub](https://docs.github.com/en/get-started/quickstart/create-a-repo)
+
+Of course, if you want to just see the code feel free [to view it on GitHub](https://github.com/oktadev/okta-node-heroku-app-example).
+
+## Create your Okta application
+
+{% include setup/cli.md type="web" framework="node"
+   loginRedirectUri="http://localhost:3000/authorization-code/callback" signup="false" %}
+
+## Create your Express application
+
+Now it's time to turn your attention to writing your application.  You will scaffold your application using `express application generator`.  Open the folder where your application will live and run the following command:
+
+```console
+npx express-generator
+```
+
+### Install your dependencies
+
+Run the following command to install the default dependencies from express-generator:
+
+```console
+npm install
+```
+
+You will need to install some additional dependencies for your application.  
+
+First, you will add `dotenv` to house your sensitive and environment-specific information.  
+
+```console
+npm i dotenv@16.0.0
+```
+
+Next, you will add bootstrap.  You will use the bootstrap libraries in your `jade` templates for writing the front end.  Bootstrap is a simple UI framework with many samples to develop HTML pages quickly.
+
+```console
+npm i bootstrap@5.1.3
+```
+
+You will need `oidc-middleware` from Okta to help secure your application.  This package will make integrating with Okta painless, quick, and secure.  You will be able to configure the middleware with your Okta application details and start securing your routes with only a few lines of code.  `express-session` is required when using the `oidc-middleware` so you will need to install that as well.
```

Comment:
```suggestion
You will need `oidc-middleware` from Okta to help secure your application.  This package makes integrating with Okta painless, quick, and secure.  You will be able to configure the middleware with your Okta application details and start securing your routes with only a few lines of code.  `express-session` is required when using the `oidc-middleware` so you will need to install that as well.
```
adding variation in tense to avoid repetitive sentence structure